Dockerfile

```diff
@@ -129,24 +137,111 @@ ARG PHYLUM_API_URI
+ENV DOTNET_NOLOGO=1
+
+# Copy only Python packages to limit the image size. This includes the `phylum` package with it's
+# `phylum-ci` and `phylum-init` entry points, plus the `pip`, `pipenv`, and `poetry` required tools.
 COPY --from=builder ${PHYLUM_VENV} ${PHYLUM_VENV}
 
-RUN set -eux; \
+# Specify the shell options here, based on the use of pipelines.
+# Ref: https://github.com/hadolint/hadolint/wiki/DL4006
+SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
+RUN \
+    # Install pre-requisites and package manager versions for `npm`, `bundle`, and `mvn` tools
     apt-get update; \
     apt-get upgrade --yes; \
-    apt-get install --yes --no-install-recommends git; \
-    chmod +x ${PHYLUM_VENV}/bin/entrypoint.sh; \
-    phylum-init -vvv --phylum-release ${CLI_VER:-latest} --global-install; \
+    apt-get install --yes --no-install-recommends \
+        git \
+        npm \
+        bundler \
+        maven \
+        curl \
+        unzip \
+        jq \
+        ; \
+    # Make ENTRYPOINT alternative script available
+    chmod +x "${PHYLUM_VENV}/bin/entrypoint.sh"; \
+    #
+    # Install Phylum CLI
+    phylum-init -vvv --phylum-release "${CLI_VER:-latest}" --global-install; \
+    #
+    # Install and enable `corepack` with cached instances of the latest major versions of `yarn` and `pnpm` tools
+    npm install -g corepack@latest; \
+    corepack pack yarn@stable pnpm@latest; \
+    corepack enable yarn pnpm; \
+    #
+    # Create an "alias" for `curl` with secure and common options
+    printf '#!/bin/bash\ncurl --proto "=https" --tlsv1.2 -sSfL "$@"\n' > /usr/bin/curls; \
+    chmod +x /usr/bin/curls; \
+    #
+    # Manual install of `gradle`
+    GRADLE_VERSION=$(curls https://services.gradle.org/versions/current | jq -r '.version'); \
+    GRADLE_DL="https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip"; \
+    GRADLE_DL_SHA256=$(curls "${GRADLE_DL}.sha256"); \
+    curls -o gradle.zip "${GRADLE_DL}"; \
+    printf "%s *gradle.zip" "${GRADLE_DL_SHA256}" | sha256sum -c -; \
+    mkdir /opt/gradle; \
+    unzip -d /opt/gradle gradle.zip; \
+    mv /opt/gradle/gradle-"${GRADLE_VERSION}"/* /opt/gradle/; \
+    rm gradle.zip && rmdir "/opt/gradle/gradle-${GRADLE_VERSION}/"; \
+    #
+    # Manual install of `go`
+    # Ref: https://pkg.go.dev/golang.org/x/website/internal/dl
+    GO_DL_URL="https://go.dev/dl/?mode=json"; \
+    GO_DL_REL=$(curls "${GO_DL_URL}" | jq '.[0].files[] | select(.os==env.TARGETOS and .arch==env.TARGETARCH)'); \
+    GO_DL_FILENAME=$(echo "${GO_DL_REL}" | jq -r '.filename'); \
+    GO_DL_SHA256=$(echo "${GO_DL_REL}" | jq -r '.sha256'); \
+    curls -o go.tgz "https://go.dev/dl/${GO_DL_FILENAME}"; \
+    printf "%s *go.tgz" "${GO_DL_SHA256}" | sha256sum -c -; \
+    rm -rf /usr/local/go; \
+    tar -C /usr/local -xzf go.tgz; \
+    rm go.tgz; \
+    #
+    # Manual install of Rust to get `cargo` tool
+    curls https://sh.rustup.rs | sh -s -- -y --profile minimal; \
```

Comment:
Is there a reason why we're not using the package for Rust?

Reply:
The full breakdown on differences between the manual install vs. the Debian package can be found [in this comment](https://github.com/phylum-dev/roadmap/issues/380#issuecomment-1721961195), repeated here since that is in a private repo:

Adding `cargo` to the image was attempted several ways:

* Installing with the `curl` command suggested on the "Install Rust" page: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y`
  * Image size ballooned to 1.55GB
  * Adding the `--profile minimal` flag brought the size down to 953MB
* Installing with the [Debian `cargo` package](https://packages.debian.org/bookworm/cargo)
  * Image size is 1.01GB
  * This is larger when viewed independently but perhaps there will be some overall savings based on package dependencies that get installed and are shared with some of the other required tools

UPDATE: Installing manually, with all the other packages accounted for, is still smaller...by ~20MB. It also allows for the current version of rust/cargo (1.72.0) vs. the older package manager version (1.63.0).